### PR TITLE
print 'could not reach agent...' only in case of network error

### PR DESCRIPTION
**Pull request by misteriaud**
### What does this PR do?

This PR modify the returned error from `IPCEndpoint.doGet()` function for a more relevant one. 

### Motivation

Currently the `IPCEndpoint.doGet()` function used to make send request from the CLI to the daemon is returning the error
```
could not reach agent: [error]
Make sure the agent is running before requesting the runtime configuration and contact support if you continue having issues
```
even if the CLI was able to reach the Agent, but it returned an HTTP error for whatever reason.

I think this error should be returned only when the Agent is truly unreachable (Agent is not running, network error, ...).

### Describe how to test/QA your changes

Build an agent and run it. Then run the agent `status` command, which would continue to work correctly.
Stop the Agent, and rerun the `status` command, it should print the `could not reach agent` error.

### CODE REVIEW

File: pkg/api/util/ipc_endpoint.go
Comment (Kaderinho) [inline pkg/api/util/ipc_endpoint.go:63]: **💬 suggestion:** ‏I believe you can do that 

```suggestion
	    if netErr, ok := err.(*net.OpError); ok {
```

**Review by Kaderinho — APPROVED** [review]

**Comment by Kaderinho** [comment]
Do you have example of scenario were we would have a better error output ? If yes, could you please put it in the QA instructions 🙏

File: pkg/api/util/ipc_endpoint.go
Comment (pgimalac) [reply pkg/api/util/ipc_endpoint.go:63]: This is not quite equivalent as `errors.As` will unwrap errors, while the typing conversion will only check the top level one

Comment (misteriaud) [reply pkg/api/util/ipc_endpoint.go:63]: I tried what you suggested but it's not making the desirable behaviour, it's not matching the if statement. I think it's because the errors in go are stored in a tree-like structure and in this case it's not possible to cast it. The `As` method instead is searching for the provided type in this tree structure. (ref: https://pkg.go.dev/errors#As) @pierreguilmin can you confirm what I'm saying ?

Comment (pgimalac) [reply pkg/api/util/ipc_endpoint.go:63]: I think you tagged the wrong Pierre haha

Comment (pgimalac) [reply pkg/api/util/ipc_endpoint.go:63]: But about your message yeah that's also my guess, so you should probably stick to `errors.As`

Comment (misteriaud) [reply pkg/api/util/ipc_endpoint.go:63]: @pgimalac (this time it's the good one) I answered before I saw you were already in the thread ^^'

**Comment by misteriaud** [comment]
I set the target branch to #24649 because one use case will be when the `agent status component [name]` will be used but the component name is unknown.
I will wait until #24649 is merged to main to merge this one maybe ? @Kaderinho

**Pull request merged**
